modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/util/MuleExtensionUtils.java

```diff
@@ -835,4 +842,76 @@ public static Optional<ImplementingTypeModelProperty> extractImplementingTypePro
   public static <T extends NamedObject> T getNamedObject(List<T> elemenets, String name) {
     return elemenets.stream().filter(elem -> elem.getName().equals(name)).findFirst().get();
   }
+
+  /**
+   * Returns the version of Java that are supported by <b>ALL</b> the {@code extensionModels}.
+   *
+   * @param extensionModels the {@link ExtensionModel} instances to evaluate
+   * @return a {@link Set} of Java versions or an empty one if the extensions have none in common
+   * @see {@link JavaConstants}
+   * @since 4.5.0
+   */
+  public static Set<String> getCommonSupportedJavaVersions(Collection<ExtensionModel> extensionModels) {
+    List<ExtensionModel> models = extensionModels instanceof List
+        ? (List<ExtensionModel>) extensionModels
+        : new ArrayList<>(extensionModels);
+
+    Set<String> commonVersions = new LinkedHashSet<>();
+    Set<String> processedVersions = new HashSet<>();
+
+    for (int i = 0; i < models.size(); i++) {
+      for (String version : models.get(i).getSupportedJavaVersions()) {
+        if (!processedVersions.add(version)) {
+          continue;
+        }
+
+        boolean isCommon = true;
+        for (int j = 0; j < models.size(); j++) {
+          if (j == i) {
+            continue;
+          }
+
+          if (!models.get(j).getSupportedJavaVersions().contains(version)) {
+            isCommon = false;
+            break;
+          }
+        }
+
+        if (isCommon) {
+          commonVersions.add(version);
+        }
+      }
+    }
```

Comment:
why not just use `Stream.distinct()`?

Reply:
because this is not the same. What this is doing is looking the java versions that are common to all extensions. distinct() on its down doesn't fix this